### ISSUE TITLE
Fix keyboard interactions for dialogs

### ISSUE THREAD

**Issue by jasongrout** [jupyterlab/jupyterlab#1001]
Using one of the dialogs with a keyboard is terrible right now:
- [ ] Focused items should be indicated visually - remove all the `outline:none` CSS. See https://github.com/jupyter/jupyterlab/issues/911 for suggestions on this.
- [ ] Tabbing should focus items in the dialog: right now buttons set the tabIndex to -1.
- [ ] Tabbing should not focus items outside the modal. It seems that we have several options here:
  - remove the tabIndex for every item outside the dialog while the dialog is up
  - capture tab/shift-tab on the first/last controls in the dialog and explicitly set the focus to cycle only in controls in the dialog
  - capture tab/shift-tab in the entire dialog, and handle all focus

See also https://github.com/jupyter/jupyterlab/issues/911 for other accessibility issues.

**Comment by annabu** [comment]
Here are a few more reads on getting the modals accessible in terms of what keyboard keys should do and what aria labels to use:
- https://www.w3.org/TR/wai-aria-practices/#dialog_modal
- https://www.marcozehe.de/2015/02/05/advanced-aria-tip-2-accessible-modal-dialogs/

**Comment by ellisonbg** [comment]
@cameronoelsen this one is also related to the dialog design...
